frontend/src/components/dashboard/dashboard.tsx

```diff
@@ -29,10 +47,22 @@ const Dashboard: FC = () => {
     if (user) {
       dispatch(userDetailsActions.getUserDetails());
     }
-    dispatch(dashboardActions.getCourses({ title: '', categoryKey: '' }));
     dispatch(dashboardActions.getCategories());
   }, [dispatch]);
 
+  useEffect(() => {
+    dispatch(
+      dashboardActions.getCourses({
+        title: title ?? '',
+        categoryKey: category ?? '',
+        page: pageFromParams
+          ? Number(pageFromParams)
+          : PaginationDefaultValue.DEFAULT_PAGE,
```

Comment:
Instead of providing default values for page in two places, it looks like you could provide default when you are getting it from params